icu4c/source/tools/genrb/reslist.cpp

```diff
@@ -1692,3 +1696,50 @@ SRBRoot::compactStringsV2(UHashtable *stringSet, UErrorCode &errorCode) {
     // +1 to account for the initial zero in f16BitUnits
     assert(f16BitUnits.length() <= (f16BitStringsLength + 1));
 }
+
+void SResource::applyFilter(
+        const PathFilter& /*filter*/,
+        ResKeyPath& /*path*/,
+        const SRBRoot* /*bundle*/) {
+    // Only a few resource types (tables) are capable of being filtered.
+}
+
+void TableResource::applyFilter(
+        const PathFilter& filter,
+        ResKeyPath& path,
+        const SRBRoot* bundle) {
+    SResource* prev = nullptr;
+    SResource* curr = fFirst;
+    for (; curr != NULL;) {
+        path.push(curr->getKeyString(bundle));
+        auto inclusion = filter.match(path);
+        if (inclusion == PathFilter::EInclusion::INCLUDE) {
+            // Include whole subtree
+            // no-op
+        } else if (inclusion == PathFilter::EInclusion::EXCLUDE) {
+            // Reject the whole subtree
+            // Remove it from the linked list
+            std::cerr << "Removing subtree: " << path << std::endl;
```

Comment:
Suppress if --quiet.

Reply:
It looks like we have both --quiet and --verbose.  I guarded this with --verbose.